### PR TITLE
Overload the definitions of dataclasses.asdict and dataclasses.astuple

### PR DESCRIPTION
**Pull request by sk-**
`dataclasses.asdict` and `dataclasses.astuple` have a factory argument, that defaults to `dict` and `tuple` respectively.

Without this overload, the inferred type when using the default argument is `<nothing>` and the error in case of wanting to get the items of a dict is:

```
error: <nothing> has no attribute "items"
```

### CODE REVIEW

**Comment by sk-** [comment]
Any ideas why I'm getting the error:

```main.py:1: error: Cannot find 'builtins' module. Typeshed appears broken! (diff)```

when running some of the `mypy_selftest`?

**Comment by ilevkivskyi** [comment]
Or maybe this https://github.com/python/typeshed/blob/master/tests/mypy_selftest.py#L19 should be updated from `mypy/typeshed` to `mypy/mypy/typeshed`?

File: stdlib/3.7/dataclasses.pyi [hunk 1]
```diff
@@ -6,8 +6,14 @@ _T = TypeVar('_T')
 class _MISSING_TYPE: ...
 MISSING: _MISSING_TYPE
 
+@overload
+def asdict(obj: Any) -> Dict: ...
+@overload
 def asdict(obj: Any, *, dict_factory: Callable[[List[Tuple[str, Any]]], _T] = ...) -> _T: ...
```
[the remark this answers is not in the document]
Comment (sk-) [reply stdlib/3.7/dataclasses.pyi]: Done.

File: stdlib/3.7/dataclasses.pyi [hunk 2]
```diff
@@ -6,8 +6,14 @@ _T = TypeVar('_T')
 class _MISSING_TYPE: ...
 MISSING: _MISSING_TYPE
 
+@overload
+def asdict(obj: Any) -> Dict: ...
```
[the remark this answers is not in the document]
Comment (sk-) [reply stdlib/3.7/dataclasses.pyi]: Done.

File: stdlib/3.7/dataclasses.pyi [hunk 3]
```diff
@@ -6,8 +6,14 @@ _T = TypeVar('_T')
...
+@overload
 def asdict(obj: Any, *, dict_factory: Callable[[List[Tuple[str, Any]]], _T] = ...) -> _T: ...
 
+@overload
+def astuple(obj: Any) -> Tuple: ...
```
[the remark this answers is not in the document]
Comment (sk-) [reply stdlib/3.7/dataclasses.pyi]: Done.

**Pull request merged**